telegram/src/test/kotlin/com/github/kotlintelegrambot/updater/ExecutorLooperTest.kt

```diff
@@ -1,61 +1,35 @@
 package com.github.kotlintelegrambot.updater
 
-import com.github.kotlintelegrambot.testutils.DirectExecutor
 import junit.framework.TestCase.assertEquals
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.runTest
 import org.junit.jupiter.api.Test
-import java.lang.RuntimeException
+import kotlin.random.Random
 
 class ExecutorLooperTest {
 
-    private val sut = ExecutorLooper(
-        loopExecutor = DirectExecutor(),
-    )
+    private fun createExecutorLooper(coroutineDispatcher: CoroutineDispatcher) =
+        ExecutorLooper(coroutineDispatcher)
 
     @Test
-    fun `loops until quit is called`() {
+    fun `loops until quit is called`() = runTest {
+        val sut = createExecutorLooper(StandardTestDispatcher(testScheduler))
         var count = 0
+        val expectedCount: Int = Random.nextInt(1000)
 
-        sut.loop {
-            count++
-
-            if (count == 13) {
-                sut.quit()
-            }
-        }
-
-        assertEquals(13, count)
-    }
-
-    @Test
-    fun `loops until thread is interrupted`() {
-        var count = 0
+        println("JcLog: Start test $expectedCount")
 
         sut.loop {
             count++
 
-            if (count == 13) {
-                Thread.currentThread().interrupt()
+            if (count == expectedCount) {
+                sut.quit()
             }
         }
+        advanceUntilIdle()
 
-        assertEquals(13, count)
-    }
-
-    @Test
-    fun `loops until an exception is thrown`() {
```

Comment:
Is this one not needed anymore? What happens when a `Exception` is thrown now?

Reply:
When an exception is thrown, the job is stopped, so maybe we can keep this test as well, but I don't think it gives us a lot of value.
Let me see if I can keep it as well

Reply:
Ahh I see, but the exception is still propagated because we are using `launch`?